src/colorizer/Collection.ts

```diff
@@ -307,25 +342,89 @@ export default class Collection {
    * Collection representing its contents (either the loaded collection or a dummy collection
    * containing just the dataset).
    * @param url the URL resource to attempt to load.
-   * @param fetchMethod optional override for the fetch method.
+   * @param options optional configuration object containing any of the following properties:
+   *  - `fetchMethod` optional override for the fetch method.
+   *  - `reportWarning` optional callback for reporting warning messages during loading.
    * @throws an error if `url` is not a URL.
    * @returns a Promise of a new Collection object, either loaded from a collection JSON file or
    * generated as a wrapper around a single dataset.
    */
-  public static async loadFromAmbiguousUrl(url: string, fetchMethod = fetchWithTimeout): Promise<Collection> {
+  public static async loadFromAmbiguousUrl(
+    url: string,
+    options: Partial<{ fetchMethod: typeof fetchWithTimeout; reportWarning: ReportWarningCallback }> = {}
+  ): Promise<Collection> {
     // TODO: Also handle Nucmorph URLs that are pasted in? If website base URL matches, redirect?
 
     if (!isUrl(url)) {
       throw new Error(`Provided resource '${url}' is not a URL and cannot be loaded.`);
     }
 
+    let collectionLoadError: Error | null = null;
+    let datasetLoadError: Error | null = null;
     try {
-      return await Collection.loadCollection(url, fetchMethod);
+      return await Collection.loadCollection(url, options);
     } catch (e) {
+      collectionLoadError = e as Error;
+      console.warn(e);
       console.log("URL resource could not be parsed as a collection; attempting to make a single-database collection.");
     }
 
     // Could not load as a collection, attempt to load as a dataset.
-    return await Collection.makeCollectionFromSingleDataset(url);
+    try {
+      const collection = Collection.makeCollectionFromSingleDataset(url);
+      // Attempt to load the default dataset immediately to surface any loading errors.
+      const loadResult = await collection.tryLoadDataset(collection.getDefaultDatasetKey());
+      if (!loadResult.loaded) {
+        throw new Error(loadResult.errorMessage);
+      }
+      return collection;
+    } catch (e) {
+      datasetLoadError = e as Error;
+    }
+
+    // Assumption: if both loads failed and the file matches one of the default filenames,
```

Comment:
suggestion so this comment isn't so necessary: instead of returning inside of each of your `try`s you could define a `result` where you also define your errors on 362, and then if it's defined here, you could return and do your error handling in an `else`

Reply:
I also thing everything below here could be a `handleCollectionLoadError` or something